### PR TITLE
Rewrite the mod

### PR DESCRIPTION
**Pull request by Alexdoru**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -15,53 +15,135 @@ autoUpdateBuildScript = false
...
+# Enables using modern java syntax (up to version 17) via Jabel, while still targeting JVM 8.
+# See https://github.com/bsideup/jabel for details on how this works.
+enableModernJavaSyntax = true
+
+# Enables injecting missing generics into the decompiled source code for a better coding experience
+# Turns most publicly visible List, Map, etc. into proper List<Type>, Map<K, V> types
+enableGenericInjection = false
+
+# Generate a class with String fields for the mod id, name, version and group name named with the fields below
+generateGradleTokenClass = net.torocraft.Tags
```
Comment (glowredman) [inline gradle.properties]: I think you want `net.torocraft.torohealthmod.Tags`

File: gradle.properties [hunk 2]
```diff
@@ -15,53 +15,135 @@ autoUpdateBuildScript = false
...
+gradleTokenModId = GRADLETOKEN_MODID
```
Comment (glowredman) [inline gradle.properties]: Don't tokenize the modid please.

File: gradle.properties [hunk 3]
```diff
@@ -15,53 +15,135 @@ autoUpdateBuildScript = false
...
+gradleTokenModName =
+gradleTokenVersion = GRADLETOKEN_VERSION
+gradleTokenGroupName =
+# [DEPRECATED]
+# Multiple source files can be defined here by providing a comma-seperated list: Class1.java,Class2.java,Class3.java
 # public static final String VERSION = "GRADLETOKEN_VERSION";
 # The string's content will be replaced with your mod's version when compiled. You should use this to specify your mod's
 # version in @Mod([...], version = VERSION, [...])
 # Leave these properties empty to skip individual token replacements
 replaceGradleTokenInFile = ToroHealthMod.java
```
Comment (glowredman) [inline gradle.properties]: You use `generateGradleTokenClass` so clear this one. Also you'd need to reference `Tags.VERSION` in the `@Mod` annotation on the main mod class.

File: gradle.properties [hunk 4]
```diff
@@ -15,53 +15,135 @@ autoUpdateBuildScript = false
...
 # Provides setup for Mixins if enabled. If you don't know what mixins are: Keep it disabled!
-usesMixins = false
+usesMixins = true
+# Adds some debug arguments like verbose output and export
+usesMixinDebug = true
 # Specify the location of your implementation of IMixinConfigPlugin. Leave it empty otherwise.
 mixinPlugin =
 # Specify the package that contains all of your Mixins. You may only place Mixins in this package or the build will fail!
-mixinsPackage =
+mixinsPackage = mixin.mixins
```
Comment (glowredman) [inline gradle.properties]: See https://github.com/GTNewHorizons/Hodgepodge/pull/247

File: src/main/java/net/torocraft/torohealthmod/client/configuration/ConfigurationHandler.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+import net.minecraftforge.common.config.Configuration;
+import net.torocraft.torohealthmod.ToroHealthMod;
+import net.torocraft.torohealthmod.client.event.ToroHealthEventHandler;
+
+import cpw.mods.fml.client.event.ConfigChangedEvent;
+import cpw.mods.fml.common.eventhandler.SubscribeEvent;
+import cpw.mods.fml.relauncher.Side;
+import cpw.mods.fml.relauncher.SideOnly;
+
+@SideOnly(Side.CLIENT)
```
Comment (glowredman) [inline src/main/java/net/torocraft/torohealthmod/client/configuration/ConfigurationHandler.java]: Don't use `@SideOnly`

File: src/main/java/net/torocraft/torohealthmod/client/event/ToroHealthEventHandler.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import net.minecraftforge.event.entity.living.LivingEvent;
+import net.torocraft.torohealthmod.client.particle.DamageParticles;
+import net.torocraft.torohealthmod.mixin.ducks.EntityLivingBaseDuck;
+
+import cpw.mods.fml.common.eventhandler.EventPriority;
+import cpw.mods.fml.common.eventhandler.SubscribeEvent;
+import cpw.mods.fml.relauncher.Side;
+import cpw.mods.fml.relauncher.SideOnly;
+
+@SideOnly(Side.CLIENT)
```
Comment (glowredman) [inline src/main/java/net/torocraft/torohealthmod/client/event/ToroHealthEventHandler.java]: Don't use `@SideOnly`.

File: src/main/java/net/torocraft/torohealthmod/client/particle/DamageParticles.java [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+import net.minecraft.util.Vec3;
+import net.minecraft.world.World;
+import net.torocraft.torohealthmod.client.configuration.ConfigurationHandler;
+
+import org.lwjgl.opengl.GL11;
+
+import cpw.mods.fml.relauncher.Side;
+import cpw.mods.fml.relauncher.SideOnly;
+
+@SideOnly(Side.CLIENT)
```
Comment (glowredman) [inline src/main/java/net/torocraft/torohealthmod/client/particle/DamageParticles.java]: Don't use `@SideOnly`.

**Review by glowredman — CHANGES_REQUESTED** [review]

File: gradle.properties (hunk 2 above)
Comment (Alexdoru) [reply gradle.properties]: why ?

File: gradle.properties (hunk 4 above)
Comment (Alexdoru) [reply gradle.properties]: no

**Review by glowredman — APPROVED** [review]

**Pull request merged**